### PR TITLE
Speech verb options in the character setup

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Adds options in the setup to change these say verbs
- Say
- Ask
- Exclaim
- Whisper
- Yell
- Sing

image

Can be anything that doesn't exceed 25 characters.

This PR is basically speech setup lite, no speech replacers.

## Why It's Good For The Game

speech replacers are too heavy, but this is knees weaks arms spaghetti and opens up some interesting character customization

## Changelog
:cl:
add: Added say verb modification options in the character setup.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/language.dm [hunk 1]
```diff
@@ -43,4 +43,18 @@ SUBSYSTEM_DEF(language)
...
+		if(cli.prefs.ask_verb)
+			H.dna.species.ask_mod = cli.prefs.ask_verb
+		if(cli.prefs.exclaim_verb)
+			H.dna.species.exclaim_mod = cli.prefs.exclaim_verb
+		if(cli.prefs.whisper_verb)
+			H.dna.species.whisper_mod = cli.prefs.whisper_verb
+		if(cli.prefs.yell_verb)
+			H.dna.species.yell_mod = cli.prefs.yell_verb
+		if(cli.prefs.sing_verb)
+			H.dna.species.sing_mod = cli.prefs.sing_verb
```
Comment (BobJoga) [inline code/controllers/subsystem/language.dm:59]: this is in a skyrat edit section

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -136,6 +136,13 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
 	var/list/body_descriptors = list()
 
 	var/list/alt_titles_preferences = list()
+
+	var/say_verb = ""
+	var/ask_verb = ""
+	var/exclaim_verb = ""
+	var/whisper_verb = ""
+	var/yell_verb = ""
+	var/sing_verb = ""
```
Comment (BobJoga) [inline code/modules/client/preferences.dm:145]: this is in a skyrat edit section

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -399,11 +405,18 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
...
-			dat += "<BR><b>Special Names:</b><BR>"
+			dat += "<b>Say Verb:</b> <a href='?_src_=prefs;preference=say_verb;task=input'>[say_verb ? say_verb : "Default"]</a><BR>"
+			dat += "<b>Ask Verb:</b> <a href='?_src_=prefs;preference=ask_verb;task=input'>[ask_verb ? ask_verb : "Default"]</a><BR>"
+			dat += "<b>Exclaim Verb:</b> <a href='?_src_=prefs;preference=exclaim_verb;task=input'>[exclaim_verb ? exclaim_verb : "Default"]</a><BR>"
+			dat += "<b>Whisper Verb:</b> <a href='?_src_=prefs;preference=whisper_verb;task=input'>[whisper_verb ? whisper_verb : "Default"]</a><BR>"
+			dat += "<b>Yell Verb:</b> <a href='?_src_=prefs;preference=yell_verb;task=input'>[yell_verb ? yell_verb : "Default"]</a><BR>"
+			dat += "<b>Sing Verb:</b> <a href='?_src_=prefs;preference=sing_verb;task=input'>[sing_verb ? sing_verb : "Default"]</a>"
+
+			dat += "<h2>Special Names</h2>"
+			//
```
Comment (BobJoga) [inline code/modules/client/preferences.dm]: also in a skyrat edit section

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -2191,6 +2204,48 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
...
+						msg = "" //just to make sure it saves as a string even though it doesn't make much sense for byond
+					if(msg)
+						yell_verb = strip_html_simple(msg, 25, TRUE)
+
+				if("sing_verb")
+					var/msg = input(usr, "Set your ask verb. Do not abuse this.", "Sing Verb", sing_verb) as message|null
+					if(!msg)
+						msg = "" //just to make sure it saves as a string even though it doesn't make much sense for byond
+					if(msg)
+						sing_verb = strip_html_simple(msg, 25, TRUE)
```
Comment (BobJoga) [inline code/modules/client/preferences.dm:2250]: this, too, is skyrat edit

**Comment by SpaceVampire** [comment]
25 characters seems pretty long, and free-will was a mistake. I can already see people doing stupid shit with this.

**Comment by BobJoga** [comment]
> 25 characters seems pretty long, and free-will was a mistake. I can already see people doing stupid shit with this.

People will do stupid shit. But whatever.

**Comment by Arctaisia** [comment]
More customizability is always nice, I support this, although maybe 25 is a little too long, 20 might be a better option but I could be wrong!

**Comment by calibraptor** [comment]
I can already see people doing really dumb stuff with this.

Nothing admins can't enforce though.

**Pull request closed without merging**